### PR TITLE
Adds GET request for all users

### PR DESCRIPTION
**Pull request by nkytruong**
I've added an endpoint for GET /api/users

- New controller
- New model
- New tests

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -228,25 +228,52 @@ describe("PATCH /api/articles/:article_id", () => {
...
+  });
+
+  test("GET 404: returns correct error message if invalid endpoint", () => {
+    return request(app)
+      .get("/api/user")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Route Not Found");
+      });
+  });
```
Comment (MKD01) [inline __tests__/index.test.js:278]: This is a generic 404 test as this is simply an incorrect endpoint the same as /api/banana. This should not be in the describe for GET /api/users.

File: __tests__/index.test.js [hunk 2]
```diff
@@ -228,25 +228,52 @@ describe("PATCH /api/articles/:article_id", () => {
...
+          expect(user).toEqual(
+            expect.objectContaining({
+              username: expect.any(String),
+              name: expect.any(String),
+              avatar_url: expect.any(String),
+            })
+          );
+        });
+      });
+  });
```
Comment (MKD01) [inline __tests__/index.test.js:269]: Really good test

**Review by MKD01 — COMMENTED** [review]
Really good work. One small change to the testing which should be made but once that has been done feel free to merge this in without needing to send an updated pr for it.

File: __tests__/index.test.js (hunk 1 above)
Comment (nkytruong) [reply __tests__/index.test.js:278]: Should this be in the describe for GET /api instead?

**Pull request merged**